### DISCUSSION TITLE
Raising an event in a parallel state transition does not propagate to sibling parallel states?

### DISCUSSION THREAD

**Discussion by earslap** [statelyai/xstate#4456] [Q&A] [answered]
Am I misunderstanding how this is implemented in XState? In another statechart building application this works as expected, so I wonder if the semantics are different in XState? I expect s2 to go to d when I initiate a2b, as a2b raises c2d - but it doesn't.

So press simulate, then click a2b to raise the event to see: https://stately.ai/registry/editor/641775d0-f3d9-4c77-afae-ce4762496b5c?machineId=8f3ef083-f103-45ba-9e58-b026659fca20&mode=Design

I expect the final state to be s1.b, s2.d but instead it is s1.b, s2.c

**Comment by davidkpiano** [comment, marked as answer]
This is only because the simulator currently doesn't execute actions. This will be fixed soon.
